cou/steps/hypervisor.py

```diff
@@ -145,8 +157,10 @@ def azs(self) -> AZs:
             for unit in app.units.values():
                 # NOTE(rgildein): If there is no AZ, we will use empty string and all units will
                 #                 belong to a single group.
-                az = unit.machine.az or ""
-                azs[az].app_units[app.name].append(unit)
+                machine = unit.machine
+                if machine in self.machines:
```

Comment:
I think we should log skipped machine, so I'm maybe doing something like this:

```python
for unit in app.units.values():
    if unit.machine not in self.machines:
        logger.debug("skipping machine %s", unit.machine)
        continue

    az = ...
```
What do you say?